### PR TITLE
Label regex schema check [FLYW-1405]

### PR DESCRIPTION
**Pull request by ryandavisflywheel**

### CODE REVIEW

File: flywheel_common/providers/provider.py [hunk 1]
```diff
@@ -1,19 +1,22 @@
 """Base provider class for all provider types"""
+import re
 from abc import abstractmethod
 from marshmallow import Schema, fields, ValidationError, validate
 
-
 from .. import errors
 
+LABEL_REGEX = re.compile('^[a-z]{1,24}$')
```
Comment (ehlertjd) [inline flywheel_common/providers/provider.py]: Are we safe to assume that numbers are OK as well? I could see them being useful, at least after the first character.

Comment (ryandavisflywheel) [reply flywheel_common/providers/provider.py]: In short, that would be OK.  We talked about this last week (https://flywheelio.atlassian.net/browse/FLYW-1405) so I'd have to change what we decided to add numbers then go verify in other repos that things will still work as expected.  
...

A much longer answer:
If we wanted to make this change it follows that this label is used as the "cloud_prefix" for a GCP compute instance name in Condor and the name is "common_name" in Condor's Ansible:
```
  vars:
    suffix: "{{ ansible_date_time.iso8601_micro | to_uuid }}"
    common_name: "{{ cloud_prefix | replace('-', '') }}-{{ suffix.split('-')[0] }}"
```
which winds up looking something like `mylabel-a9775564`

 The "common_name" must adhere to the regex added here https://github.com/flywheel-io/condor/pull/98: 
`[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?|[1-9][0-9]{0,19}`

so numbers would be fine ... also Condor puts everything in the label to lowercase before using it as the "cloud_prefix" so we could even go as far as allowing uppercase letters.

Comment (ehlertjd) [reply flywheel_common/providers/provider.py]: Lets keep it as is for now

Comment (ryandavisflywheel) [reply flywheel_common/providers/provider.py]: ah... I added them in before I saw this...

Comment (ryandavisflywheel) [reply flywheel_common/providers/provider.py]: I think everything  should be fine without changing other repos now that I've looked/thought about it

**Pull request merged**